[feat] 서브템플릿 조회 API 및 서비스 구현

yeo-li opened the pull request:
## #️⃣ 연관된 이슈
close: #46

## 🚀 작업 내용
- API: `/api/v1/templates/{templateId}subtemplates`
- [x] SubTemplateConverter.toListReponseDTO 구현
- [x] SubTemplateRepository.findByTemplate 추가
- [x] SubTemplateRepositoryFacade.findBySubTemplate 구현
- [x] SubTemplateService.getSubTemplates 구현
- [x] TemplateController.getSubTemplate 비지니스 로직 추가
- [x] 테스트 코드 작성


## 🔍 리뷰 요청 사항 및 공유자료
<!-- 리뷰어에게 확인받고 싶은 내용을 적어주세요 -->
- 부족한 유효성 검증 및 리팩토링 사항
- 테스트 코드 통과 여부

## ✅ PR Checklist
- [x] 커밋 메시지 컨벤션을 지켰습니다.
- [x] 변경 사항에 대한 테스트를 완료했습니다.

## 🤔 Review 예상 시간
<!-- 5분, 10분 등등... -->
20분+

yeo-li committed:
[feat] SubTemplatConverter toListResponseDTO 구현(#46)

yeo-li committed:
[feat] SubTemplateRepository findByTemplate 추가(#46)

yeo-li committed:
[feat] SubTemplateRepositoryFacade findBySubTemplate 메서드 구현(#46)

yeo-li committed:
[feat] SubTemplateService getSubTemplates 구현(#46)

yeo-li committed:
[refactor] SubTemplateExceptionType enum 리팩토링(#46)

yeo-li committed:
[feat] TemplateController getSubtemplates 비지니스 로직 추가(#46)

yeo-li committed:
[test] 테스트 코드 작성(#46)

soheeGit commented on backend/src/main/java/com/hotpack/krocs/domain/templates/service/SubTemplateServiceImpl.java:
```diff
@@ -51,4 +52,32 @@ public SubTemplateCreateResponseDTO createSubTemplates(Long templateId,
+            if (template == null) {
+                throw new SubTemplateException(
+                    SubTemplateExceptionType.SUB_TEMPLATE_TEMPLATE_NOT_FOUND);
+            }
```
findByTemplateId 메서드 안에서도 template == null 검증을 하고 있는 것 같습니다!
혹시 둘이 다른 에러이기 때문에 넣으신 것일까요?

yeo-li replied:
앗 검증하고 있는 줄은 몰랐네요..! `TEMPLATE_NOT_FOUND` 에러가 `SUBTEMPATE`에서 발생했음을 표현하고자 위 형식으로 적었습니다! 제거하도록 하겠습니다!

soheeGit commented on backend/src/main/java/com/hotpack/krocs/domain/templates/service/SubTemplateServiceImpl.java:
```diff
@@ -51,4 +52,32 @@ public SubTemplateCreateResponseDTO createSubTemplates(Long templateId,
+                    SubTemplateExceptionType.SUB_TEMPLATE_TEMPLATE_NOT_FOUND);
+            }
+
+            List<SubTemplate> subTemplates = subTemplateRepositoryFacade.findBySubTemplate(
+                template);
+            if (subTemplates.isEmpty()) {
+                throw new SubTemplateException(SubTemplateExceptionType.SUB_TEMPLATE_NOT_FOUND);
+            }
```
이부분은 제 부분도 고쳐야 할 부분이 있을 것 같아서... 팀 전체적으로 논의를 해봤으면 좋겠는데, subTemplates가 비어있으면 빈 배열로 나가는게 프론트가 처리하기에 좋지 않을까 싶어서 제안드립니다! 비어있으면 무조건 예외를 던지는 게 프론트 입장에서 부자연스러울 것 같아서요! 성열님 생각이 궁금합니다!

yeo-li replied:
저도 생각해보니 프론트에서 빈 배열이 아닌 예외를 던지면 따로 핸들링을 해주어야 할 것 같아 불편할 것 같습니다! 예외 말고 빈 배열을 던지도록 가원님과 마지막 상의 후 결정 지어보아요!

soheeGit reviewed (changes requested):
늦은시간 고생하셨습니다! 이부분들만 의견 부탁드릴게요!!

yeo-li committed:
[refactor] SubTemplateService.getSubtemplates() findByTemplatedId() 예외 처리 제거(#46)

Cycrypto commented on backend/src/main/java/com/hotpack/krocs/domain/templates/converter/SubTemplateConverter.java:
```diff
@@ -41,6 +41,14 @@ public SubTemplateCreateResponseDTO toCreateResponseDTO(
             .build();
     }
 
+    public List<SubTemplateResponseDTO> toListResponseDTO(List<SubTemplate> subTemplates) {
```
stream api를 이용하면 조금 더 깔끔할 거 같아요

```
public List<SubTemplateResponseDTO> toListResponseDTO(List<SubTemplate> subTemplates) {
    return subTemplates.stream()
                       .map(this::toResponseDTO)
                       .toList();
}
```

Cycrypto reviewed (changes requested):
고생 많으셨습니다~

yeo-li committed:
[refactor] SubTemplateServiceImpl.getSubTemplates 리팩토링(#46)

조회된 SubTemplate가 없을때 기존 코드는 SUB_TEMPLATE_NOT_FOUND를 발생시켰지만, 빈 배열을 반환하고 예외를 발생시키지 않도록 리팩토링 했습니다.

yeo-li committed:
[refactor] SubTemplateConverter.toL
 for문 -> 스트림 문법으로 변환(#46)

yeo-li committed:
[feat] 서브 템플릿(SubTemplate) 삭제 기능 구현(#51)

Cycrypto reviewed (dismissed):
이륙허가합니다
고생하셨습니다

yeo-li committed:
[test] 테스트 코드 작성(#51)

yeo-li committed:
Merge pull request #72 from seeds-hotpack/feat/#51/delete-sub-template-api

[feat] 서브템플릿 삭제 API 및 서비스 구현

gawoooon reviewed (approved):
LGTM ~

soheeGit reviewed (approved):
고생하셨습니다~~
